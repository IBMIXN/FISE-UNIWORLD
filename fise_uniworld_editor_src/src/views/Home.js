import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Grid, Button, Container, Typography } from "@material-ui/core";
import UniWorldScreenShotImg from "../assets/img/uniworldscreenshot.png";
import { Link as RouterLink } from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  jumbo: {
    paddingTop: theme.spacing(5),
    paddingBottom: theme.spacing(5),
    display: "flex",
    alignItems: "center",
    height: "80vh",
  },
  img: {
    width: "100%",
  },
  divider: {
    padding: theme.spacing(1),
  },
  viewButton: {
    "&:hover": {
      backgroundColor: theme.palette.info.dark,
    },
    backgroundColor: theme.palette.info.main,
  },
  createButton: {
    "&:hover": {
      backgroundColor: theme.palette.success.dark,
    },
    marginLeft: theme.spacing(2),
    backgroundColor: theme.palette.success.main,
  },
}));

const Home = () => {
  const classes = useStyles();

  return (
    <React.Fragment>
      <Container>
        <Grid container className={classes.jumbo}>
          <Grid item md={8} xs={12}>
            <Typography variant="h2">UniWorld Editor</Typography>
            <Typography variant="subtitle1">
              This is a web editor tool for managing virtual rooms in the{" "}
              <a href="https://brianmin.tech/uniworld/" target="_blank" rel="noreferrer noopener">
                UniWorld platform
              </a>
            </Typography>
            <div className={classes.divider} />
            <Button
              className={classes.viewButton}
              variant="contained"
              color="primary"
              size="large"
              component={RouterLink}
              to="/eventrooms"
            >
              View Event Rooms
            </Button>
            <Button
              className={classes.createButton}
              variant="contained"
              color="secondary"
              size="large"
              component={RouterLink}
              to="/create"
            >
              Create Event Room
            </Button>
          </Grid>
          <Grid item md={4} xs={12}>
            <img src={UniWorldScreenShotImg} alt="" className={classes.img} />
          </Grid>
        </Grid>
      </Container>
    </React.Fragment>
  );
};

export default Home;
