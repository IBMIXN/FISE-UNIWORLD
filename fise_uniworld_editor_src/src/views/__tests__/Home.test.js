import React from "react";
import { shallow } from "enzyme";
import Home from "../Home";

describe("Home view test", () => {
  it("should match snapshot", () => {
    const component = shallow(<Home />);
    expect(component).toMatchSnapshot();
  });
});
