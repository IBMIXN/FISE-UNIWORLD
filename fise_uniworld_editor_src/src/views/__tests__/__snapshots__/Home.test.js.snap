// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view test should match snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-jumbo-1"
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={8}
        xs={12}
      >
        <WithStyles(ForwardRef(Typography))
          variant="h2"
        >
          UniWorld Editor
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          This is a web editor tool for managing virtual rooms in the
           
          <a
            href="https://brianmin.tech/uniworld/"
            rel="noreferrer noopener"
            target="_blank"
          >
            UniWorld platform
          </a>
        </WithStyles(ForwardRef(Typography))>
        <div
          className="makeStyles-divider-3"
        />
        <WithStyles(ForwardRef(Button))
          className="makeStyles-viewButton-4"
          color="primary"
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          size="large"
          to="/eventrooms"
          variant="contained"
        >
          View Event Rooms
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-createButton-5"
          color="secondary"
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          size="large"
          to="/create"
          variant="contained"
        >
          Create Event Room
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={4}
        xs={12}
      >
        <img
          alt=""
          className="makeStyles-img-2"
          src="uniworldscreenshot.png"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
